### PR TITLE
Stop router-specific admin properties from being overridden by jobs

### PR DESCRIPTION
**Pull request by haibchen**
"azkaban.job.cluster.classpaths" and "azkaban.job.cluster.nativelib" are two critical job properties that ensure jobs are executed properly when a cluster router is enabled.

However, jobs can accidentally override these properties, resulting in job failures. For example, some HadoopJava jobs may output every one of its job properties as its output property, including these two.
When a downstream job starts, it gets them as its own job properties (nodes gets output properties from their in-nodes), which may be incorrect.

These two are meant to be generated by the cluster router per job and stay private within a job.
Here we essentially always allow the router to generate "azkaban.job.cluster.classpaths" and "azkaban.job.cluster.nativelib"

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java [hunk 1]
```diff
@@ -399,6 +403,15 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
     }
   }
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java:405]: Can you please add JavaDoc?

File: azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java [hunk 2]
```diff
@@ -381,7 +381,11 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
       // inject cluster jars and native libraries into jobs through properties
       // User's job props should take precedence over cluster props
       Props finalProps = getClusterSpecificJobProps(targetCluster, jobProps, pluginLoadProps);
+      Props nonOverriddableClusterProps = getClusterSpecificNonOverridableJobProps(finalProps);
       finalProps.putAll(jobProps);
+      // add router-specific properties that are critical for job execution as the last step
```
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java:386]: Please add cautionary comment, perhaps something in all caps to make sure no one adds props after that code.

**Review by djaiswal83 — CHANGES_REQUESTED** [review]
This is a good catch!
I have couple of minor comments.

File: azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java (hunk 1 above)
Comment (haibchen) [reply azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java:412]: added

File: azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java (hunk 2 above)
Comment (haibchen) [reply azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java]: addressed

**Review by djaiswal83 — APPROVED** [review]
Thanks for addressing the comments.

**Pull request merged**